config/application_space_test.go

```diff
@@ -0,0 +1,321 @@
+				AppSpace: &configpb.ApplicationSpace{
+					Name: "like-real-appspace-name",
+				},
+			}
+
+			mockClient.EXPECT().
+				ReadApplicationSpace(
+					gomock.Any(),
+					test.WrapMatcher(test.EqualProto(req)),
+					gomock.Any(),
+				).Return(beResp, nil)
+
+			resp, err := configClient.ReadApplicationSpace(ctx, req)
+			Expect(err).To(Succeed())
+			Expect(resp).To(test.EqualProto(beResp))
+		})
+	})
+	Describe("AppSpaceCreate", func() {
+		DescribeTable("Request error during Create ",
+			func(req *configpb.CreateApplicationSpaceRequest, causeMatcher, msgMatcher, codeMatcher OmegaMatcher) {
+				resp, err := configClient.CreateApplicationSpace(ctx, req)
+				Expect(err).To(HaveOccurred())
+				Expect(resp).To(BeNil())
+
+				var clientErr *sdkerrors.ClientError
+				Expect(errors.As(err, &clientErr)).To(BeTrue(), "is client error")
+				Expect(clientErr.Unwrap()).To(causeMatcher)
+				Expect(clientErr.Message()).To(msgMatcher)
+				Expect(clientErr.Code()).To(codeMatcher)
+			},
+			Entry("nil request", nil, BeNil(), Equal("invalid nil request"), Equal(codes.InvalidArgument)),
+		)
+
+		It("Create", func() {
+			displayNamePb := &wrapperspb.StringValue{Value: "Like real AppSpace Name"}
+			req := &configpb.CreateApplicationSpaceRequest{
+				CustomerId:  "gid:like-real-customer-id",
+				Name:        "like-real-appspace-name",
+				DisplayName: displayNamePb,
+			}
+			beResp := &configpb.CreateApplicationSpaceResponse{
+				Id:   "gid:like-real-app_space-id",
+				Etag: "123qwe",
+			}
+
+			mockClient.EXPECT().CreateApplicationSpace(
+				gomock.Any(),
+				test.WrapMatcher(test.EqualProto(req)),
+				gomock.Any(),
+			).Return(beResp, nil)
+
+			resp, err := configClient.CreateApplicationSpace(ctx, req)
+			Expect(err).To(Succeed())
+			Expect(resp).To(test.EqualProto(beResp))
+		})
+	})
+	Describe("AppSpaceUpdate", func() {
+		DescribeTable("Request error during Update ",
+			func(req *configpb.UpdateApplicationSpaceRequest, causeMatcher, msgMatcher, codeMatcher OmegaMatcher) {
+				resp, err := configClient.UpdateApplicationSpace(ctx, req)
+				Expect(err).To(HaveOccurred())
+				Expect(resp).To(BeNil())
+
+				var clientErr *sdkerrors.ClientError
+				Expect(errors.As(err, &clientErr)).To(BeTrue(), "is client error")
+				Expect(clientErr.Unwrap()).To(causeMatcher)
+				Expect(clientErr.Message()).To(msgMatcher)
+				Expect(clientErr.Code()).To(codeMatcher)
+			},
+			Entry("nil request", nil, BeNil(), Equal("invalid nil request"), Equal(codes.InvalidArgument)),
+		)
+
+		It("Update", func() {
+			displayNamePb := &wrapperspb.StringValue{Value: "Like real AppSpace Name"}
+			etagPb := &wrapperspb.StringValue{Value: "123qwert"}
+			req := &configpb.UpdateApplicationSpaceRequest{
+				Id:          "gid:like-real-app-space-id",
+				Etag:        etagPb,
+				DisplayName: displayNamePb,
+			}
+			beResp := &configpb.UpdateApplicationSpaceResponse{
+				Id:   "gid:like-real-app_space-id",
+				Etag: "123qwert",
+			}
```

Comment:
If you update with DisplayName and response doesn't contain display name, it seems kind a weird.

Reply:
DisplayName is not in the response.
Since everything is mocked, how can I read after mocked updated to check if really updated?